src/matrix.jl

```diff
@@ -0,0 +1,71 @@
+istable(::Type{<:AbstractMatrix}) = false
+
+rows(m::T) where {T <: AbstractMatrix} = throw(ArgumentError("a '$T' is not a table; see `?Tables.table` for ways to treat an AbstractMatrix as a table"))
+columns(m::T) where {T <: AbstractMatrix} = throw(ArgumentError("a '$T' is not a table; see `?Tables.table` for ways to treat an AbstractMatrix as a table"))
+
+struct MatrixTable{T}
+    names::Vector{Symbol}
+    lookup::Dict{Symbol, Int}
+    matrix::T
+end
+
+istable(::Type{<:MatrixTable}) = true
+names(m::MatrixTable) = getfield(m, 1)
+
+# row interface
+struct MatrixRow{T}
+    row::Int
+    source::MatrixTable{T}
+end
+
+Base.getproperty(m::MatrixRow, ::Type, col::Int, nm::Symbol) =
+    getfield(getfield(m, 2), 3)[getfield(m, 1), col]
+Base.getproperty(m::MatrixRow, nm::Symbol) =
+    getfield(getfield(m, 2), 3)[getfield(m, 1), getfield(getfield(m, 2), 2)[nm]]
+Base.propertynames(m::MatrixRow) = names(getfield(m, 2))
+
+rowaccess(::Type{<:MatrixTable}) = true
+schema(m::MatrixTable{T}) where {T} = Schema(Tuple(names(m)), NTuple{size(getfield(m, 3), 2), eltype(T)})
+rows(m::MatrixTable) = m
+Base.eltype(m::MatrixTable{T}) where {T} = MatrixRow{T}
+Base.length(m::MatrixTable) = size(getfield(m, 3), 1)
+
+function Base.iterate(m::MatrixTable, st=1)
+    st > length(m) && return nothing
+    return MatrixRow(st, m), st + 1
+end
+
+# column interface
+columnaccess(::Type{<:MatrixTable}) = true
+columns(m::MatrixTable) = m
+Base.getproperty(m::MatrixTable, ::Type{T}, col::Int, nm::Symbol) where {T} = getfield(m, 3)[:, col]
+Base.getproperty(m::MatrixTable, nm::Symbol) = getfield(m, 3)[:, getfield(m, 2)[nm]]
+Base.propertynames(m::MatrixTable) = names(m)
+
+"""
+Tables.table(m::AbstractMatrix; header::Vector{Symbol}=...)
+
+Convert an AbstractMatrix (Matrix, Adjoint, etc.) to a MatrixTable, which satisfies the Tables.jl interface. This allows accesing the matrix via `Tables.rows` and `Tables.columns`. An optional keyword argument `header` can be passed as a Vector{Symbol} to be used as the column names. Note that no copy of the AbstractMatrix is made.
```

Comment:
Missing backticks around names. I'd also say "wrap" instead of "convert".

Also, how about breaking lines at 92 chars? Text is almost impossible to read currently.